### ISSUE TITLE
[v1.1.0 rc0][v1.0.2] MDB_BAD_VALSIZE: Unsupported size of key/DB name/data, or wrong DUPFIXED size

### ISSUE THREAD

**Issue by Plong-Wasin** [meilisearch/meilisearch#3574]
**Describe the bug**
I use thai language to search but it error "MDB_BAD_VALSIZE: Unsupported size of key/DB name/data, or wrong DUPFIXED size".
My dataset format is [{"product_id":xxxxx,"product_title":"yyyyyy"}] "yyyyyy" is thai language
My setting index is default setting
I can send simple data to you

**To Reproduce**
1. Import data
2. Use meilisearch dashboard
3. Open browser devtools
4. Search "วริ" , "น้"
5. See error

**Expected behavior**
Indexation process completes successfully.

**Screenshots**
![image](https://user-images.githubusercontent.com/72617740/223930199-3d4b3bf3-31a1-41f9-935c-057d8d2ae1a8.png)
![image](https://user-images.githubusercontent.com/72617740/223930236-478839f4-f432-45a1-a1c2-61db3e6e8a74.png)
![image](https://user-images.githubusercontent.com/72617740/223932901-588be1cc-fc34-4f82-a17a-359aeb5a0ad2.png)

**Meilisearch version:**
[v1.0.2](https://github.com/meilisearch/meilisearch/releases/tag/v1.0.2)
[v1.1.0 rc0](https://github.com/meilisearch/meilisearch/releases/tag/v1.1.0-rc.0)

**Additional context**
[Docker official image](https://hub.docker.com/r/getmeili/meilisearch)

**Comment by Kerollmops** [comment]
Hey @Plong-Wasin 👋,

Thank you very much for this bug report. Could you please send us your dataset in this issue or via email to kero@meilisearch.com? It is an internal issue that must not be thrown and be correctly handled by the engine.

Have a nice day!

**Comment by Kerollmops** [comment]
Hey @Plong-Wasin,

I did receive your email with the dataset and was able to reproduce the bug.

I found out that the error [originates from this line](https://github.com/meilisearch/meilisearch/blob/df3986cd83589519f3acd93f304045642089ca08/milli/src/search/criteria/exactness.rs#L347). I think the Charabia tokenizer is returning an empty string when it parses the input query and tries to search for it in the internal database, however, [LMDB returns an error](https://twitter.com/Kerollmops/status/1633769488356106240) when we use an empty string to get a value. We will fix that in the next v1.1.0 release.

Thank you very much for your help!
